src/library/vm/vm_aux.cpp

```diff
@@ -56,13 +56,40 @@ vm_obj vm_try_for(vm_obj const &, vm_obj const & n, vm_obj const & thunk) {
     }
 }
 
+vm_obj vm_try_for_time (vm_obj const &, vm_obj const & n, vm_obj const & thunk) {
```

Comment:
You also need to provide a (dummy / always throwing an exception / always executing the thunk) implementation for the javascript version.
```suggestion
vm_obj vm_try_for_time (vm_obj const &, vm_obj const & n, vm_obj const & thunk) {
#if defined(LEAN_MULTI_THREAD)
```